arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -468,28 +473,81 @@ std::pair<irs::string_ref, irs::string_ref> splitAnalyzerName( // split name
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// @brief validate that features are supported by arangod an ensure that their
-///        dependencies are met
+/// @brief read analyzers from vocbase
+/// @return visitation compleated fully
 ////////////////////////////////////////////////////////////////////////////////
-arangodb::Result validateFeatures(irs::flags const& features) {
-  for(auto& feature: features) {
-    if (&irs::frequency::type() == feature) {
-      // no extra validation required
-    } else if (&irs::norm::type() == feature) {
-      // no extra validation required
-    } else if (&irs::position::type() == feature) {
-      if (!features.check(irs::frequency::type())) {
-        return arangodb::Result( // result
-          TRI_ERROR_BAD_PARAMETER, // code
-          std::string("missing feature '") + std::string(irs::frequency::type().name()) +"' required when '" + std::string(feature->name()) + "' feature is specified"
-        );
-      }
-    } else if (feature) {
+arangodb::Result visitAnalyzers( // visit analyzers
+  TRI_vocbase_t& vocbase, // vocbase to visit
+  std::function<arangodb::Result(arangodb::velocypack::Slice const& slice)> const& visitor // visitor
+) {
+  if (arangodb::ServerState::instance()->isClusterRole()) {
```

Comment:
Should this be `isCoordinator()` instead?
Does the `_analyzers` collection exist on DB servers at all? Its name there is probably `s100xxx` or something.

Reply:
this should be coordinator or db-server, as both of them should gather their data from the entire cluster